lib/postgrex/simple_connection.ex

```diff
@@ -337,7 +337,12 @@ defmodule Postgrex.SimpleConnection do
   @impl :gen_statem
   def handle_event(type, content, statem_state, state)
 
-  def handle_event(:internal, {:connect, _}, @state, %{state: {mod, mod_state}} = state) do
+  def handle_event(:internal, {:connect, :reconnect}, @state, %{protocol: protocol} = state) do
+    Protocol.disconnect(:reconnect, protocol)
```

Comment:
Hi, thanks for this fix.

Could it be though that this needs to consider a case in which `protocol` is `nil`?

We got this error in production with the latest `postgrex` `0.19.3`

![image](https://github.com/user-attachments/assets/2a999d50-1cb8-4485-ba26-31b48d2f487a)

Not `cancel_request(nil)` call.

Thanks!

Reply:
These are the logs from when this error happened...
![image](https://github.com/user-attachments/assets/1924913b-d076-42a6-ae57-c919450408ff)

Reply:
Pushed to main.

Reply:
Thanks @josevalim :pray: